### PR TITLE
Docs: to-do lists Markdown update.

### PR DESCRIPTION
**Pull request by godai78**
Closes: #8879  
  
Apparently also closes: #8887

Adding information about the `[ ]` shortcode to the Autoformatting and To-do lists guides.

### CODE REVIEW

**Comment by Mgsy** [comment]
@godai78, I've added the to-do list plugin to the autoformatting demo.

**Comment by godai78** [comment]
It works not, great thanks!

**Comment by AnnaTomanek** [comment]
Can you change the target to `stable` and merge `stable` into this first, just in case :wink: ?

**Comment by godai78** [comment]
So it is done as requested.

File: packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js [hunk 1]
```diff
@@ -39,6 +41,7 @@ ClassicEditor
 				'blockQuote',
 				'codeBlock',
 				'horizontalLine',
+				'todolist',
```
Comment (AnnaTomanek) [inline packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js:44]: I'd move it to other lists, just like it is in the to-do list guide. Feels more natural.

File: packages/ckeditor5-list/docs/features/todo-lists.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ order: 20
 
 # To-do lists
 
-The {@link module:list/todolist~TodoList to-do list} feature lets you create a list of interactive checkboxes with labels. It supports all features of regular lists so you can nest a to-do list together with {@link features/lists bulleted and numbered lists} in any combination.
+The {@link module:list/todolist~TodoList to-do list} feature lets you create a list of interactive checkboxes with labels. It supports all features of regular lists so you can nest a to-do list together with {@link features/lists bulleted and numbered lists} in any combination. To-do lists can be introduced using toolbar buttons, or with Markdown code provided by the {@link features/autoformat autoformatting feature}, simply start a line with `[ ]` followed by a space.
```
Comment (AnnaTomanek) [inline packages/ckeditor5-list/docs/features/todo-lists.md:9]:
```suggestion
The {@link module:list/todolist~TodoList to-do list} feature lets you create a list of interactive checkboxes with labels. It supports all features of regular lists so you can nest a to-do list together with {@link features/lists bulleted and numbered lists} in any combination. To-do lists can be introduced using the toolbar button. Thanks to the integration with the {@link features/autoformat autoformatting feature}, they can also be added with Markdown code &mdash; simply start a line with `[ ]` followed by a space.
```

File: packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js (hunk 1 above)
Comment (godai78) [reply packages/ckeditor5-autoformat/docs/_snippets/features/autoformat.js]: Done.

**Review by AnnaTomanek — APPROVED** [review]
Thanks!

**Pull request merged**